jina/peapods/zmq.py

```diff
@@ -46,7 +45,7 @@ def __init__(self, args: 'argparse.Namespace', logger: 'logging.Logger' = None):
         """
         self.args = args
         self.name = args.name or self.__class__.__name__
-        self.logger = logger or get_logger(self.name, **vars(args))
```

Comment:
it always receives the logger from the outside, and having 2 owners of this logger could make it hard to release the handlers and close them